### PR TITLE
Better quickstart

### PR DESCRIPTION
**Pull request by WardLT**
# Description

Make the introduction simpler, and a quick introduction to configuring Parsl to the Quickstart.

My goal was to make it easier for people coming from a "Python Multiprocessing" background to understand how to use Parsl. 
The document has reams of useful information already, so I just created a simple walkthrough to teach people where to look. 

Things (potentially) left to do:
- [x] Add documentation pages about the `parsl.concurrent` features. At least in the Bag-of-Tasks
- [x] Explain where a Parsl program should run (head node vs batch script)
- [x] ~Make a "tutorial" for configuration. I find more people confused about that than anything else~ [Subject of another PR]
- [x] Give the same snappy makeover we gave to rtd to the README.rst used by GitHub. The current one is text heavy
- [x] Make/find a figure for the Quickstart

## Type of change

Choose which options apply, and delete the ones which do not apply.

- Documentation update

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,40 +1,56 @@
...
-
-* Concurrent execution of tasks in a bag-of-tasks program.
-* Procedural workflows in which tasks are executed following control logic.
-* Parallel dataflow in which tasks are executed when their data dependencies are met.
-* Many-task applications in which many computing resources are used to perform various computational tasks.
-* Dynamic workflows in which the workflow is dynamically determined during execution.
-* Interactive parallel programming through notebooks or interactive.
+Parsl is parallel programming library for Python for parallelism beyond a single computer.
+
+At the easiest level, you can use Parsl to run functions just like Python's executors but across *multiple computers*:
```
Comment (danielskatz) [inline docs/index.rst:6]: maybe "multiple cores and nodes" would be better?

File: docs/index.rst [hunk 2]
```diff
@@ -1,40 +1,56 @@
...
+    from parsl.configs.htex_local import config
+
+    with ParslPoolExecutor(config) as exec:
+        results = pool.map(function, [...])
+
+Start with the `configuration quickstart <quickstart.html#getting-started>`_ to learn how to tell Parsl how to use your computing resource.
+Parsl can run on most supercomputers and with major cloud providers.
+
+The real power of Parsl comes is expressing multi-step workflows of functions.
+Parsl lets you chain functions together and will launch each function as soon as possible.
```
Comment (danielskatz) [inline docs/index.rst:20]: "as soon as the inputs and computing resources are available"?

File: docs/quickstart.rst [hunk 3]
```diff
@@ -49,11 +49,29 @@ The conda documentation provides `instructions <https://docs.conda.io/projects/c
...
+Task execution is brokered by a `"Data Flow Kernel" <#benefits-of-a-data-flow-kernel>`_ that runs on your local system.
+
+We describe these components briefly here, and link to more details in the `User Guide <userguide/index.html>`_.
+
+Application Types
+^^^^^^^^^^^^^^^^^
+
+Parsl enables concurrent execution of Python functions (``python_app``)
+or external applications (``bash_app``).
+The logic for both are described by Python functions marked with with Parsl decorators.
```
Comment (danielskatz) [inline docs/quickstart.rst:71]:
```suggestion
The logic for both are described by Python functions marked with Parsl decorators.
```

File: docs/quickstart.rst [hunk 4]
```diff
@@ -78,10 +96,65 @@ with hello world Python and Bash apps.
...
+Learn more about the types of Apps and their options `here <userguide/apps.html>`_.
+
+Executors
+^^^^^^^^^
+
+Executors define how Parsl deploys work on a computer.
+Many types are available, each with different advantages.
+
+The :class:`~parsl.executors.high_throughput.executor.HighThroughputExecutor` is most familiar to most people.
+Like Python's ``ProcessPoolExecutor``, the workers it creates are separate Python processes.
```
Comment (danielskatz) [inline docs/quickstart.rst:112]: for previous two lines, perhaps
```The :class:`~parsl.executors.high_throughput.executor.HighThroughputExecutor`, like Python's ``ProcessPoolExecutor``, creates workers that are separate Python processes.

File: docs/quickstart.rst [hunk 5]
```diff
@@ -78,10 +96,65 @@ with hello world Python and Bash apps.
...
+Often, this simply involves specifying the correct Python environment and
+(described below) how to launch the Executor on each acquired computers.
+
+Learn more about Providers `here <userguide/execution.html#execution-providers>`_ and
+find examples for common supercomputers `here <userguide/configuring.html>`_.
+
+Launchers
+^^^^^^^^^
+
+The Launcher defines how to spread workers across all computers available in a Block.
```
[the remark this answers is not in the document]
Comment (WardLT) [reply docs/quickstart.rst:138]: Yea, that does feel antiquated and, perhaps, imprecise. How about "nodes?"

File: docs/quickstart.rst [hunk 6]
```diff
@@ -78,10 +96,65 @@ with hello world Python and Bash apps.
...
+Execution Providers
+^^^^^^^^^^^^^^^^^^^
+
+Resource providers allow Parsl to gain access to computing power.
+For supercomputers, gaining resources often requires requesting them from a scheduler (e.g., Slurm).
+Parsl Providers write the requests to requisition **"Blocks"** of computers (e.g., supercomputer nodes) on your behalf.
+Parsl comes pre-packaged with Providers compatible with most supercomputers and some cloud computing services.
+
+Another key role of Providers is defining how to start an Executor on a remote computer.
+Often, this simply involves specifying the correct Python environment and
```
[the remark this answers is not in the document]
Comment (WardLT) [reply docs/quickstart.rst:129]: Totally

File: docs/quickstart.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielskatz) [reply docs/quickstart.rst:138]: I'm really not sure what the right word is - this is part of why this is hard to explain to others easily - no standard terms that work across all cases

[the remark this answers is not in the document]
Comment (WardLT) [reply docs/quickstart.rst:138]: We call them nodes in many providers, so I'll stick with that

File: docs/quickstart.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (WardLT) [reply docs/quickstart.rst:129]: I opted to add Config to a wrap-up section at the end, as I couldn't figure out how to do it without first introducing launchers.

File: README.rst [hunk 7]
```diff
@@ -15,7 +15,7 @@ Parsl can be used to implement various parallel computing paradigms:
 * Dynamic workflows in which the workflow is determined during execution
 * Interactive parallel programming through notebooks or another interactive mechanism
 
-The latest Parsl version available on PyPi is v1.1.0.
+The latest Parsl version available on PyPi is v1.3.0.
```
Comment (benclifford) [inline README.rst:18]: not any more! there's an automatic release process and nothing to update any version numbers outside of the release. see eg https://github.com/Parsl/parsl/issues/2463

**Review by benclifford — APPROVED** [review]

**Pull request merged**